util/push/push.go

```diff
@@ -55,7 +59,26 @@ func Push(ctx context.Context, sm *session.Manager, cs content.Provider, dgst di
 	if insecure {
 		opt.PlainHTTP = insecure
 	}
-
+	if customCA != "" {
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			rootCAs = x509.NewCertPool()
+		}
+		certs, err := ioutil.ReadFile(customCA)
```

Comment:
Shouldn't this be read from client session?

Reply:
My idea was to specify a file located in buildkitd's filesystem namespace to add to the ca certificate chain.

Then I can just specify that for example `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt` should be used to verify the registry. This is the default location where k8s mounts its internal ca cert to.

Reply:
I think client should read `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt` and expose it to the daemon via the session RPC.

I don't think we should allow client to access arbitrary file path on the daemon.

Reply:
Can you point me to some documentation on how to expose files (which are not part of the build context) to the exporter? I'll update the PR accordingly.